packages/client/src/components/form/IDVerification/Banner.tsx

```diff
@@ -59,14 +65,41 @@ const ConfirmationModal: React.FC<{
 export const IDVerificationBanner = ({
   type,
   idFieldName,
-  setFieldValue
+  setFieldValue,
+  form
 }: {
   type: BannerType
   setFieldValue: (name: string, value: IFormFieldValue) => void
   idFieldName: string
+  form: IFormSectionData
 }) => {
   const intl = useIntl()
   const [modal, openModal] = useModal()
+  const { declarationId = '', pageId: section } = useParams()
+  const declaration = useDeclaration(declarationId)
+  const userId = useSelector(getUserDetails)?.id
+  useEffect(() => {
+    if (
+      (type === 'authenticated' || type === 'failedFetchIdDetails') &&
+      !!form[idFieldName] &&
+      !!declaration &&
+      section &&
+      userId
+    ) {
+      // update and save the declaration in indexedDB
+      // to persist the ID verification details
+      // we have to do it here because the updated values of entire section
+      // of the fields depending on id field are not available
+      // when the link button is rendered
+      writeDeclarationByUserWithoutStateUpdate(userId, {
+        ...declaration,
+        data: {
+          ...declaration.data,
+          [section]: merge(declaration.data[section], form)
+        }
+      })
+    }
+  }, [declaration, form, idFieldName, section, type, userId])
```

Comment:
When would this trigger? Any time the form updates and the form section has a ID Verification banner, we would write the form to IndexedDB? Why wouldn't we do this on the link button onClick?

Reply:
Thanks @naftis good catch. Well, we indeed write declaration on the link button click. That persists the declaration to the state it was right before the click, so that  when the user returns to the form, can see the form values. 

But this write declaration call is used to store, after the identity details are fetched. By then the link button is hidden and authenticated banner is rendered. This had to be persisted because of the replace back button reload [issue](https://github.com/opencrvs/opencrvs-core/issues/9096#issuecomment-2804381010).

Reply:
Oh okay, thanks, makes sense. So next question, why won't we write to store in the place where we fetch the identity details?

Reply:
I first tried to do so, right beneath this line https://github.com/opencrvs/opencrvs-core/blob/1b80260da12c497f8bb97cbcce83ecabe0e868d7/packages/client/src/components/form/LinkButton.tsx#L67

But when `setFieldValue` is called with a data value, the link button gets hidden by the form conditional, before the writing is finished.

Reply:
Hmm.. difficult. I'm just thinking would the bug reoccur if a country derived a form field value from the REDIRECT but they are not using the verification banner

Reply:
That's a good concern. Agreed, it is kind of a twisted logic...

I thought if our form would somehow have an ability to execute functions on value change... But retracted from there thinking it of a big change

Reply:
Would this bug happen if the back button used the previous section of the form, instead of browser back? I'm not suggesting a change, but I think this is how Events v2 handle the navigation, and therefore we could just add a comment like `HOTFIX for Events v1 National ID integration: In events v2 the back button will navigate to the previous section instead of using browser back".

I'm not sure! We can brainstorm this tomorrow in the handover call as well.